### PR TITLE
Add a minimal settings database interface.

### PR DESCRIPTION
**Pull request by mrstegeman**

### CODE REVIEW

File: gateway_addon/database.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+_DB_PATHS = [
+    os.path.join(os.path.expanduser('~'),
+                 'mozilla-iot',
+                 'gateway',
+                 'db.sqlite3'),
+    os.path.join(os.path.expanduser('~'),
+                 '.mozilla-iot',
+                 'config',
+                 'db.sqlite3'),
```
Comment (hobinjk) [inline gateway_addon/database.py:16]: Is there any reason this can't include a relative path given that addons are almost always run from `build/addons`?

Comment (mrstegeman) [reply gateway_addon/database.py:16]: I typically run out of tree when I'm developing add-ons. Also, relative paths won't be reliable if the add-ons change their current directory. There's an optional `path` parameter that can be passed to `Database()` that should let devs do whatever they want, though.

Comment (hobinjk) [reply gateway_addon/database.py:16]: My main constraint is that I want any python add-on I download to work out of the box on my laptop where the gateway is in a slightly nonstandard location. The unified config location fixes this for the future but if adding '../../../db.sqlite3' works for 90% of custom setups I think we should have it.

Comment (mrstegeman) [reply gateway_addon/database.py:16]: Fair enough. Done.

**Review by hobinjk — APPROVED** [review]

**Comment by dhylands** [comment]
We already have an environment variable for config-editor to locate the database. Perhaps we should try to have some consistency?

**Comment by mrstegeman** [comment]
@dhylands that environment variable is included in the search paths.

**Pull request merged**